### DISCUSSION TITLE
Writing external command to be nu-ready without structured output?

### DISCUSSION THREAD

**Discussion by Jiehong** [nushell/nushell#10768] [Q&A]
Hello,

Today, I see 2 options for external commands to be integrated with nu:

- write a nu plugin;
- if the command has structured output, use that and parse it with open/ write a wrapper.

What if I want to write a cli tool in like python or java, but I want to make it compatible with nu and able to return structured output if in nu, or stdout output if in bash or something.

How can one handle such a thing? (if at all possible)

**Comment by fdncred** [comment]
You can write plugins in python or java but to have them interact with nushell they'd have to be registered and named nu_plugin_some_name. However you may be able to write that code in such a way that it detects whether the parent shell is nushell or not and do what you want. You could rename the file by also symlinking it to a different name. I'm just guessing here. I don't think anyone has tried this that I know of.
